File: siserp-logistica-backend/src/main/java/com/besoft/siserp/lgtc/service/UbigeoService.java
package com.besoft.siserp.lgtc.service;

import java.util.List;

import com.besoft.siserp.lgtc.dto.OutResponse;
import com.besoft.siserp.lgtc.dto.request.UbigeoDistritoListarRequest;
import com.besoft.siserp.lgtc.dto.request.UbigeoDistritoListarXUbigeoRequest;
import com.besoft.siserp.lgtc.dto.request.UbigeoProvinciaListarRequest;
import com.besoft.siserp.lgtc.dto.request.UbigeoProvinciaListarXUbigeoRequest;
import com.besoft.siserp.lgtc.dto.response.UbigeoDepartamentoListarResponse;
import com.besoft.siserp.lgtc.dto.response.UbigeoDistritoListarResponse;
import com.besoft.siserp.lgtc.dto.response.UbigeoProvinciaListarResponse;

public interface UbigeoService {

	public OutResponse<List<UbigeoDepartamentoListarResponse>> listarDepartamento();

	public OutResponse<List<UbigeoProvinciaListarResponse>> listarProvincia(UbigeoProvinciaListarRequest req);

	public OutResponse<List<UbigeoDistritoListarResponse>> listarDistrito(UbigeoDistritoListarRequest req);

	public OutResponse<List<UbigeoProvinciaListarResponse>> listarProvinciaXUbigeo(UbigeoProvinciaListarXUbigeoRequest req);

	public OutResponse<List<UbigeoDistritoListarResponse>> listarDistritoXUbigeo(UbigeoDistritoListarXUbigeoRequest req);
}

File: siserp-logistica-frontend/src/main/siserp-logistica-frontend/src/app/modules/intranet/services/ubigeo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { UbigeoDistritoListarRequest } from '../dto/request/ubigeo-distrito-listar.request';
import { UbigeoDistritoListarXUbigeoRequest } from '../dto/request/ubigeo-distrito-listar_x_ubig.request';
import { UbigeoProvinciaListarRequest } from '../dto/request/ubigeo-provincia-listar.request';
import { UbigeoProvinciaListarXUbigeoRequest } from '../dto/request/ubigeo-provincia-listar_x_ubig.request';
import { OutResponse } from '../dto/response/out.response';
import { UbigeoDepartamentoListarResponse } from '../dto/response/ubigeo-departamento-listar.response';
import { UbigeoDistritoListarResponse } from '../dto/response/ubigeo-distrito-listar.response';
import { UbigeoProvinciaListarResponse } from '../dto/response/ubigeo-provincia-listar.response';

@Injectable()
export class UbigeoService {

  constructor(private http: HttpClient) { }

  public listarDepartamento(): Observable<OutResponse<UbigeoDepartamentoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDepartamentoListarResponse[]>>(`${environment.logisticaBackendUrl}/ubigeo/listarDepartamento`, {});
  }

  public listarProvincia(req: UbigeoProvinciaListarRequest): Observable<OutResponse<UbigeoProvinciaListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoProvinciaListarResponse[]>>(`${environment.logisticaBackendUrl}/ubigeo/listarProvincia`, req);
  }

  public listarDistrito(req: UbigeoDistritoListarRequest): Observable<OutResponse<UbigeoDistritoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDistritoListarResponse[]>>(`${environment.logisticaBackendUrl}/ubigeo/listarDistrito`, req);
  }

  public listarProvinciaXUbigeo(req: UbigeoProvinciaListarXUbigeoRequest): Observable<OutResponse<UbigeoProvinciaListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoProvinciaListarResponse[]>>(`${environment.logisticaBackendUrl}/ubigeo/listarProvinciaXUbigeo`, req);
  }

  public listarDistritoXUbigeo(req: UbigeoDistritoListarXUbigeoRequest): Observable<OutResponse<UbigeoDistritoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDistritoListarResponse[]>>(`${environment.logisticaBackendUrl}/ubigeo/listarDistritoXUbigeo`, req);
  }
}
